examples/createContract/dapp/src/createContract.sol

```diff
@@ -2,9 +2,30 @@ pragma solidity >=0.5.15;
 
 contract Kid {
   address public parent;
+  string  public constant name = 'Kid V1';
+  bytes32 public ID;
+  bool    public initialized;
 
   constructor() public {
-    parent = msg.sender;
+      parent = msg.sender;
+      uint chainId;
+      assembly {
+          chainId := chainid
+      }
+      ID = keccak256(
```

Comment:
Is this here to do something with `chainId` and `address(this)` in the constructor?